tests/adapter/dbt/tests/adapter/materialized_views/test_on_configuration_change.py

```diff
@@ -0,0 +1,173 @@
+from typing import List
+
+import pytest
+
+from dbt.contracts.graph.model_config import OnConfigurationChangeOption
+from dbt.contracts.results import RunStatus
+from dbt.tests.util import run_dbt_and_capture, relation_from_name
+
+from dbt.tests.adapter.materialized_views.base import Base
+
+
+class OnConfigurationChangeBase(Base):
+
+    on_configuration_change: OnConfigurationChangeOption
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"on_configuration_change": str(self.on_configuration_change)}}
+
+    def assert_proper_scenario(
+        self,
+        results,
+        logs,
+        status: RunStatus,
+        messages_in_logs: List[str],
+        messages_not_in_logs: List[str],
+        rows_affected: int,
```

Comment:
on `skip or fail` scenarios would we expect the `rows_affected` to matchup? since nothing should be happening on the full refresh